.travis.yml

```diff
@@ -2,14 +2,12 @@ language: java
 
 sudo: false
 
-script:
-  - jdk_switcher use oraclejdk7
-  - mvn clean verify -P jdk7
-  - jdk_switcher use oraclejdk8
-  - mvn clean verify -P jdk8
+jdk:
+  - oraclejdk8
+  - oraclejdk7
 
-after_success:
-  - mvn clean test jacoco:report coveralls:report
+script:
```

Comment:
> How is this configured currently, am I only allowed to merge when all checks pass?

I'm not sure if Github prevents the merge, but it certainly shows failed checks. No idea how to configure this.

> Then it would be nice to directly see the distinction between failing tests and missing coverage.

If you click on *Show all checks" you should be able to see all three results.

> Not sure how this works with github, there must also be some kind of special remote ref.

Indeed. I have the following piece in my `~/.gitconfig` to fetch remote pull request refs from stash and github.

```
[remote "origin"]
    push = HEAD
    fetch = +refs/pull-requests/*/from:refs/remotes/origin/pull-request/source/*
    fetch = +refs/pull-requests/*/merge:refs/remotes/origin/pull-request/merge/*
    fetch = +refs/pull/*/head:refs/remotes/origin/pull-request/source/*
    fetch = +refs/pull/*/merge:refs/remotes/origin/pull-request/merge/*
    fetch = +refs/heads/*:refs/remotes/origin/*
```

Not sure, but I'd assume the following setup for travis would work, i.e. it would make the build pass even if coverage failed, and it should run the whole test suite only once.

```yaml
script:
  - mvn clean verify jacoco:report
after_success:
  - mvn coveralls:report  
```